components/image-builder-mk3/pkg/resolve/resolve.go

```diff
@@ -79,14 +84,56 @@ func (sr *StandaloneRefResolver) Resolve(ctx context.Context, ref string, opts .
 	}
 
 	nref := pref.String()
+	pref = reference.TrimNamed(pref)
 	span.LogKV("normalized-ref", nref)
 
-	res, _, err = r.Resolve(ctx, nref)
+	res, desc, err := r.Resolve(ctx, nref)
 	if err != nil && strings.Contains(err.Error(), "not found") {
 		err = ErrNotFound
+		return
+	}
+	fetcher, err := r.Fetcher(ctx, res)
+	if err != nil {
+		return
+	}
+
+	in, err := fetcher.Fetch(ctx, desc)
+	if err != nil {
+		return
+	}
+	defer in.Close()
+	buf, err := ioutil.ReadAll(in)
+	if err != nil {
+		return
+	}
+
+	var mf ociv1.Manifest
+	err = json.Unmarshal(buf, &mf)
+	if err != nil {
+		return "", fmt.Errorf("cannot unmarshal manifest: %w", err)
+	}
+
+	if mf.Config.Size != 0 {
+		pref, err = reference.WithDigest(pref, mf.Config.Digest)
+		return pref.String(), nil
+	}
+
+	var mfl ociv1.Index
+	err = json.Unmarshal(buf, &mfl)
+	if err != nil {
+		return
+	}
+
+	var dgst digest.Digest
+	for _, mf := range mfl.Manifests {
+		if fmt.Sprintf("%s-%s", mf.Platform.OS, mf.Platform.Architecture) == "linux-amd64" {
+			dgst = mf.Digest
+			pref, err = reference.WithDigest(pref, dgst)
+			return pref.String(), nil
+		}
```

Comment:
nit: align happy path to the left

```suggestion
	var dgst digest.Digest
	for _, mf := range mfl.Manifests {
		if fmt.Sprintf("%s-%s", mf.Platform.OS, mf.Platform.Architecture) == "linux-amd64" {
			dgst = mf.Digest
			break
		}
	}
	if dgst == "" {
		return "", fmt.Errorf("no manifest for platform linux-amd64 found")
	}
	
	pref, err = reference.WithDigest(pref, dgst)
	return pref.String(), err
```